snuba/subscriptions/executor_consumer.py

```diff
@@ -247,6 +277,34 @@ def __init__(
             self.__metrics, "executor.concurrent.clickhouse"
         )
 
+    @property
+    def max_concurrent_queries(self) -> int:
+        return self.__max_concurrent_queries
+
+    def __calculate_max_concurrent_queries(
+        self,
+        partitions: Mapping[Partition, int],
+        partition_count: int,
+        total_concurrent_queries: int,
+        use_calculated_value: bool,
+    ) -> int:
+        if (
+            partition_count <= 0 or partition_count < len(partitions)
+        ) and not use_calculated_value:
+            logger.info(f"Partition count {partition_count} wouldve been invalid")
+            return 1
+
+        if partition_count <= 0:
+            raise InvalidPartitionCount(
+                f"Partition count {partition_count} must be a non-zero positive integer."
+            )
+        if partition_count < len(partitions):
+            raise InvalidPartitionCount(
+                f"Partition count {partition_count} cannot be lower than partitions received {len(partitions)}."
+            )
+
+        return int(total_concurrent_queries / (partition_count / len(partitions))) or 1
```

Comment:
I think it is safer to round up here rather than down. Otherwise we would be potentially unnecessarily returning an error when ClickHouse would be fine with it. If we mistakenly let too many queries through, ClickHouse will error but then either way would have been an error anyway. You also don't need the `or 1` hack to avoid it being 0 if you round up.

Reply:
okay i can use `math.ciel` on this to round up. You are also saying then to forget this check:

 `partition_count < len(partitions)` 

right?